src/Raven.Server/Smuggler/Documents/Handlers/SmugglerHandler.cs

```diff
@@ -399,6 +400,96 @@ public Task GetImport()
+            {
+                if (HttpContext.Request.HasFormContentType == false)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+                    using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                    {
+                        context.Write(writer, new DynamicJsonValue
+                        {
+                            ["Type"] = "Error",
+                            ["Error"] = "Import from csv requires form content type"
+                        });
+                        return;
+                    }
+                }
+                var token = new OperationCancelToken(Database.DatabaseShutdown);
+                var result = new SmugglerResult();
+                var operationId = Database.Operations.GetNextOperationId();
+                await Database.Operations.AddOperation(Database, $"Import to: {Database.Name} from csv file", Raven.Server.Documents.Operations.Operations.OperationType.DatabaseImportFromCsv,
+                    onProgress =>
+                    {
+                        return Task.Run(async () =>
+                        {
+                            try
+                            {
+                                var reader = new MultipartReader(MultipartRequestHelper.GetBoundary(MediaTypeHeaderValue.Parse(HttpContext.Request.ContentType),
+                                    MultipartRequestHelper.MultipartBoundaryLengthLimit), HttpContext.Request.Body);
+                                while (true)
+                                {
+                                    var section = await reader.ReadNextSectionAsync().ConfigureAwait(false);
+                                    if (section == null)
+                                        break;
+
+                                    if (ContentDispositionHeaderValue.TryParse(section.ContentDisposition, out ContentDispositionHeaderValue contentDisposition) == false)
+                                        continue;
+
+                                    if (MultipartRequestHelper.HasFileContentDisposition(contentDisposition))
+                                    {
+                                        if (ContentDispositionHeaderValue.TryParse(section.ContentDisposition, out contentDisposition) == false)
+                                            continue;
+
+                                        var filename = contentDisposition.FileName.ToString().Trim('\"');
+                                        var options = new DatabaseSmugglerOptionsServerSide();
+                                        if (section.Headers.ContainsKey("Content-Encoding") && section.Headers["Content-Encoding"] == "gzip")
+                                        {
+                                            using (var gzipStream = new GZipStream(section.Body, CompressionMode.Decompress))
```

Comment:
IIRC we have utility method from getting that?

Reply:
again exactly like in smuggler import

Reply:
I don't know to what utility method you're referring to nor could i locate one even when you told me it exists, could you point me to where it reside?

Reply:
I meant something here: https://github.com/ravendb/ravendb/blob/4b2c605ffa3af88a35ae66b828f5612414a8e3f6/src/Raven.Server/Web/RequestHandler.cs#L77